syntax = "proto3";

package message;

service VerifyService {
  rpc GetVerifyCode (GetVerifyReq) returns (GetVerifyRsp) {}
}

message GetVerifyReq {
  string email = 1;
}

message GetVerifyRsp {
  int32 error = 1;
  string email = 2;
  string code = 3;
}

service StatusService {
    // ChatServer 注册请求
    rpc RegisterChatServer (RegisterChatServerRequest) returns (RegisterChatServerResponse);

    // ChatServer 断开连接请求
    rpc UnregisterChatServer (UnregisterChatServerRequest) returns (UnregisterChatServerResponse);

    // GateServer 获取 ChatServer 的 IP 和端口请求
    rpc GetChatServer (GetChatServerRequest) returns (GetChatServerResponse);

    // 用户登录请求，生成 token
    rpc Login (LoginRequest) returns (LoginResponse);
}

// ChatServer 注册请求
message RegisterChatServerRequest {
    string ip = 1;
    int32 port = 2;
}

// ChatServer 注册响应
message RegisterChatServerResponse {
    int32 error = 1;
}

// ChatServer 断开连接请求
message UnregisterChatServerRequest {
    string ip = 1;
    int32 port = 2;
}

// ChatServer 断开连接响应
message UnregisterChatServerResponse {
    int32 error = 1;
}

// GateServer 获取 ChatServer 的 IP 和端口请求
message GetChatServerRequest {
    string username = 1; // 用户名，用于标识请求用户
}

// GateServer 获取 ChatServer 的 IP 和端口响应
message GetChatServerResponse {
    int32 error = 1;
    string ip = 2;
    int32 port = 3;
    string token = 4; // 返回生成的 token
}

// 用户登录请求
message LoginRequest {
    string username = 1;
    string token = 2;
}

// 用户登录响应
message LoginResponse {
    int32 error = 1;
    string username = 2;
    string token = 3;
}